### PR TITLE
RavenDB-13581: Pass query elapsed time on LazyQueryOperation

### PR DESCRIPTION
**Pull request by garayx**

### CODE REVIEW

File: src/Raven.Client/Documents/Session/Operations/Lazy/LazyQueryOperation.cs [hunk 1]
```diff
@@ -56,12 +56,12 @@ public void HandleResponse(GetResponse response)
                     queryResult.DurationInMs = -1; // taken from cache
             }
 
-            HandleResponse(queryResult);
+            HandleResponse(queryResult, response.Elapsed);
         }
 
-        private void HandleResponse(QueryResult queryResult)
+        private void HandleResponse(QueryResult queryResult, TimeSpan responseElapsed)
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Session/Operations/Lazy/LazyQueryOperation.cs]: rename to `duration`

File: src/Raven.Client/Documents/Session/Operations/QueryOperation.cs [hunk 2]
```diff
@@ -273,6 +273,21 @@ public void EnsureIsAcceptableAndSaveResult(QueryResult result)
 
             EnsureIsAcceptable(result, _indexQuery.WaitForNonStaleResults, _sp, _session);
 
+            SaveQueryResult(result);
+        }
+
+        public void EnsureIsAcceptableAndSaveResult(QueryResult result, TimeSpan responseElapsed)
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Session/Operations/QueryOperation.cs]: rename parameter to `duration`

File: src/Raven.Client/Documents/Session/Operations/QueryOperation.cs [hunk 3]
```diff
@@ -331,6 +346,16 @@ public static void EnsureIsAcceptable(QueryResult result, bool waitForNonStaleRe
             }
         }
 
+        public static void EnsureIsAcceptable(QueryResult result, bool waitForNonStaleResults, TimeSpan? elapsed, InMemoryDocumentSessionOperations session)
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Session/Operations/QueryOperation.cs]: rename to `duration` and call this method from the other `EnsureIsAcceptable

File: src/Raven.Client/Documents/Session/Operations/QueryOperation.cs (hunk 2 above)
Comment (ppekrol) [reply src/Raven.Client/Documents/Session/Operations/QueryOperation.cs]: Maybe change to `internal` and make `duration` nullable? Then from the other `EnsureIsAcceptableAndSaveResult` call this method, but pass `null` for duration

File: src/Raven.Client/Documents/Session/Operations/Lazy/LazyQueryOperation.cs (hunk 1 above)
Comment (garayx) [reply src/Raven.Client/Documents/Session/Operations/Lazy/LazyQueryOperation.cs]: done

File: src/Raven.Client/Documents/Session/Operations/QueryOperation.cs (hunk 2 above)
Comment (garayx) [reply src/Raven.Client/Documents/Session/Operations/QueryOperation.cs]: done

**Pull request merged**